src/lib.rs

```diff
@@ -37,8 +38,14 @@ pub fn gobject_gen(input: TokenStream) -> TokenStream {
 
     match result {
         Ok(token_stream) => {
-            let output = rustfmt(&token_stream).unwrap();
-            println!("{}", output);
+            let mut config: rustfmt::config::Config = Default::default();
+            config.set().error_on_line_overflow(false);
+            let summary = rustfmt::run(rustfmt::Input::Text(token_stream.as_str().into()),
+                                       & config);
+            if !summary.has_no_errors() {
+                println!("{:#?}", summary);
+                panic!("cannot generate gobjects")
+            }
             token_stream.parse().unwrap()
```

Comment:
AFAICT rustfmt.run() modifies the passed string. The only output is a Summary struct and is only implements Debug (not Display) so it's not worth printing out unless there is an error.